_summaries/bahdanau2014neural_2.md

```diff
@@ -1,19 +1,70 @@
+
+$$c_{i} = \sum_{j=1}^{T_x} \alpha_{i j} h_{j} \qquad \boldsymbol{\alpha}_i = softmax(\mathbf{e}_{i}) \qquad e_{ij} = a(s_{i-1},h_{j})$$
+
+where the last equation represents the parameters of an _alignment model_ that learns to assign an importance to inputs around $$j$$ for the output at $$i$$. This model $$a(\cdot)$$ is jointly trained with the other parts of the encoder-decoder architecture. This allows the model to search for positions in the input sequence with the most relevant information (captured in the context-vectors) for predicting next word.
+
+
+
+## Benefits of proposed model
+
+* With BiRNN and a soft alignment (using attention) over the annotations of the input sequence, the model is able to look at various parts of the source sequence before predicting the next output word. This improves the quality of the translation.
+* By having a context vector for each word in the output, it automatically takes care of cases where source and target sequences are of different lengths.
+* The model no longer requires to encode long sequences into fixed-length vectors which allows a better encoding of the information around all words in the long sequence.
+
+
+
+## Performance
+
+The proposed model RNNsearch is evaluated on English-to-French translation task and compared against the baseline RNNencdec model. The authors train 2 versions of each model - one with input sequences of length up to 30 and another with length up to 50. The architecture includes 1000 hidden units each in the decoder and encoder RNNs. In the BiRNN encoder of RNNsearch, each RNN has 1000 units. Other hyper-parameter choices are as follows:
+  * Optimizer: SGD with Adadelta
+  * Batch size: 80 sentences
+  * Training time: 5 days
+  * Inference: Beam search
+
+<img width="400" alt="bahdanau2014neural_2b" src="bahdanau2014neural_2b.png">
+<img width="500" alt="bahdanau2014neural_2c" src="bahdanau2014neural_2c.png">
+
+
+The table above compares the performance of the models on the test set of 3003 sentences as measured by the **BLEU score**. The last column corresponds to testing on sentences which do not contain any words not seen during training. RNNsearch-50* model is trained for much longer.
+
+* The translation model with attention clearly outperforms the baseline model by a substantial margin. 
+* In fact, the performance of the RNNsearch-30 is better than RNNencdec-50 which shows the strength of RNNsearch in encoding long sequences.
+* In the _No UNK_ case, the performance of the attention-based NMT model is also comparable with a conventional phrase-based machine translator (Moses) given that Moses uses an additional monolingual corpus of 418M words. 
+* The plot illustrates the superiority of RNNsearch as the sequence length increases. The performance of the baseline model deteriorates but RNNsearch-50 is stable even beyond the length of 50.
+
```

Comment:
This is a very well done review so I struggled to find much wrong with it.  Perhaps condense the explanation of the decoder slightly and explain more the prior implementations of encoder-decoder networks.

Reply:
Thanks! The prior implementations are the same as the 2 papers previous to this one for this lecture. They are going to be summarized and discussed there, so I didn't go too deep into that and explained the proposed architecture in detail.